Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM mesosphere/mesos:0.21.1-1.1.ubuntu1404
+
+ENV HDFS_MESOS_VERSION 0.1.0
+
+COPY hdfs-mesos-$HDFS_MESOS_VERSION.tgz /hdfs/
+RUN cd /hdfs && tar xfz hdfs-mesos-$HDFS_MESOS_VERSION.tgz
+
+ENV MESOS_NATIVE_JAVA_LIBRARY /usr/local/lib/libmesos.so
+ENV MESOS_NATIVE_LIBRARY /usr/local/lib/libmesos.so
+
+WORKDIR /hdfs/hdfs-mesos-$HDFS_MESOS_VERSION
+
+CMD ./bin/hdfs-mesos
```

Comment:
After looking into this I determined that there is a fair amount of work in the ConfigServer to make it read from the system properties instead of the config file. Doing a FROM and building your own image to change the mesos-site.xml file seems like the best way for now.